onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -3813,10 +3837,17 @@ Status TensorrtExecutionProvider::CreateNodeComputeInfoFromGraph(const GraphView
         }
       }
 
-      // dump ep context model
+      // Give a warning that ep context need to be regenerated
       if (dump_ep_context_model_ && ep_context_embed_mode_) {
-        UpdateCtxNodeModelEngineContext(model_proto_.get(), reinterpret_cast<char*>(serialized_engine->data()), serialized_engine->size());
-        DumpCtxModel(model_proto_.get(), ctx_model_path_);
+        if (!is_subgraph) {
```

Comment:
Why checking is_subgraph?

Reply:
I use is_subgraph to indicate if this subgraph is the entire graph/model, if so, it means the entire graph is running on tensorrt, then it's going to be 1 ep context node. Let me add this to comment and let know if you have any ideas to make it clearer